stan/math/prim/mat/fun/cholesky_decompose.hpp

```diff
@@ -25,12 +30,26 @@ Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> cholesky_decompose(
     const Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>& m) {
   check_square("cholesky_decompose", "m", m);
   check_symmetric("cholesky_decompose", "m", m);
+#ifdef STAN_OPENCL
+  if (m.rows() > 1800) {
+    matrix_gpu m_gpu(m);
+    Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic> m_chol(m.rows(), m.cols());
+    cholesky_decompose(m_gpu, floor(m_gpu.rows() * 0.03));
```

Comment:
How is the `min_block` argument derived here?

Reply:
See comment below ( on shortcircuiting).